### PR TITLE
Product review details

### LINKED ISSUE #1253: Reviews tab: review details view
Implement the single review details view, adapting the current view that renders a notification, and integrate with Yosemite to submit updates to the review status 

Parent issue: #1160

### PR DESCRIPTION
**Pull request by ctarda**
Fixes #1253 

Another long PR, but this is basically duplicating the existing NotificationsDetailsViewController, which will be removed when we remove the feature flag.

This is the second PR in the implementation of #1160 .

Screenshots:

Animated gifs:

## Changes
- New NotificationDetailsViewController
- Updates to ProductReviewsAction and ProductReviewsStore to support updating the Review status (approve / put on hold / trash / spam)
- Updates to the Networking Layer to support reviewer avatar.

## Testing
- Checkout the branch, run the tests
- Undo the feature flag. Because the NotificationsViewController is instantiated in a Storyboard, it will be necessary to touch `FeatureFlag.swift` to:
```
        case .reviews:
            return BuildConfiguration.current == .localDeveloper
```
and it will also be necessary to change the class of the first view controller in the Notifications.storyboard from NotificationsViewController to ReviewsViewController
- As an alternative, revert locally the commit dc49cdf1e3b57d3b926923268393d73c32a64319
- Navigate to the reviews tab > select a review. 
- Play with the buttons: approve / unapprove / delete / spam, and make sure that the status is reflected and all actions can be undone. 
- It wouldn't hurt that the status of the review is propagated to the store (what I've done to test this is having the product open in wp-admin)
- Have in mind that marking a review as spam or trashing it makes it disappear from wp-admin, so you will need to be fast tapping "undo" if you want to bring it back 

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Notifications/ReviewDetailsViewController.swift [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+            ServiceLocator.analytics.track(.notificationReviewAction, withProperties: ["type": ProductReviewStatus.approved.rawValue])
+
+            self.moderateReview(siteID: reviewSiteID, reviewID: reviewID, doneStatus: .approved, undoStatus: .hold)
+        }
+
+        commentCell.onUnapprove = { [weak self] in
+            guard let self = self else {
+                return
+            }
+            ServiceLocator.analytics.track(.notificationReviewApprovedTapped)
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Notifications/ReviewDetailsViewController.swift]: might we want to track a different event like `notificationReviewUnapprovedTapped`? or just use the event below with `type` property?

File: Networking/Networking/Model/Product/ProductReview.swift [hunk 2]
```diff
@@ -91,6 +97,8 @@ private extension ProductReview {
         case status         = "status"
         case reviewer       = "reviewer"
         case reviewerEmail  = "reviewer_email"
+        case avatarURLs     = "reviewer_avatar_urls"
+        case avatar96       = "96"
```
[the remark this answers is not in the document]
Comment (ctarda) [reply Networking/Networking/Model/Product/ProductReview.swift]: Yes, the payload contains three different sizes of avatar. I went with that one, as it seemed a large enough (without being too large) and avoided creating another nested object with three different URLs, when we are not going to use them. I will add a comment to clarify 😊

File: Networking/Networking/Remote/ProductReviewsRemote.swift [hunk 3]
```diff
@@ -62,7 +62,7 @@ public final class ProductReviewsRemote: Remote {
     public func updateProductReviewStatus(for siteID: Int, reviewID: Int, statusKey: String, completion: @escaping (ProductReview?, Error?) -> Void) {
         let path = "\(Path.reviews)/\(reviewID)"
         let parameters = [ParameterKey.status: statusKey]
-        let request = JetpackRequest(wooApiVersion: .mark3, method: .put, siteID: siteID, path: path, parameters: parameters)
+        let request = JetpackRequest(wooApiVersion: .mark3, method: .post, siteID: siteID, path: path, parameters: parameters)
```
[the remark this answers is not in the document]
Comment (ctarda) [reply Networking/Networking/Remote/ProductReviewsRemote.swift]: It does not. It only works on `post`. I would not bet anything on it working previously though, and all other methods modifying content seem to be post as well 🤷‍♂

File: WooCommerce/Classes/ViewRelated/Notifications/ReviewDetailsViewController.swift [hunk 4]
```diff
@@ -0,0 +1,501 @@
...
+    /// Setup: Header Cell (Plain)
+    ///
+    func setupHeaderPlainCell(_ cell: UITableViewCell) {
+        guard let headerCell = cell as? NoteDetailsHeaderPlainTableViewCell else {
+            return
+        }
+
+        headerCell.leftImage = .productImage
+        headerCell.rightImage = .externalImage
+        headerCell.plainText = product?.name ?? "Unknown"
```
[the remark this answers is not in the document]
Comment (ctarda) [reply WooCommerce/Classes/ViewRelated/Notifications/ReviewDetailsViewController.swift]: Right, good catch!

File: WooCommerce/Classes/ViewRelated/Notifications/ReviewDetailsViewController.swift (hunk 1 above)
Comment (ctarda) [reply WooCommerce/Classes/ViewRelated/Notifications/ReviewDetailsViewController.swift]: That event corresponds to `review_detail_approve_button_tapped`, which in the spreadsheet is marked as a UI event, associated with the button. So, we send that event, that indicates that the button was tapped, and then another event that passes the type of status, in sequence:

```
                ServiceLocator.analytics.track(.notificationReviewApprovedTapped)
                ServiceLocator.analytics.track(.notificationReviewAction, withProperties: ["type": CommentStatus.unapproved.rawValue])
```

**Comment by ctarda** [comment]
Thanks folks! I'll merge as soon as CI is completed

**Comment by kyleaparker** [comment]
Some comments on the UI:

1. can we increase the margins on the single review screen:
- On product link section at the top: 12pt top and bottom
- On the review section: 16pt above the avatar and 16pt below the review actions

2. The colour of the product title, reviewer name, and date look darker than the secondary gray (#3C3C3C)

Original design for reference
ios-review

**Comment by ctarda** [comment]
Thanks @kyleaparker I've logged a separate issue for those fixes to the UI: #1303

**Pull request merged**